src/NUnitFramework/tests/Attributes/TestOrderAttributeTests.cs

```diff
@@ -22,6 +24,51 @@ public void CheckOrderIsCorrect()
             Assert.AreEqual(work.Children[1].Test.Name, "Y_SecondTest");
             Assert.AreEqual(work.Children[2].Test.Name, "Z_ThirdTest");
         }
-        
+
+        [Test]
+        [TestCaseSource(nameof(Cases))]
+        public void CheckClassOrderIsCorrect(List<Type> candidateTypes)
+        {
+            var testSuite = TestBuilder.MakeFixture(candidateTypes);
+
+            var work = TestBuilder.PrepareWorkItem(testSuite, null) 
+                as CompositeWorkItem;
+
+            var fixtureWorkItems = 
+                ((work.Children[0] as CompositeWorkItem)
+                .Children[0] as CompositeWorkItem)
+                .Children;
+
+            Assert.AreEqual(candidateTypes.Count, fixtureWorkItems.Count);
+            for (var i = 1; i < fixtureWorkItems.Count; i++)
+            {
+                var previousTestOrder = int.Parse(fixtureWorkItems[i - 1]
+                    .Test.Properties.Get(PropertyNames.Order).ToString());
+                var currentTestOrder = int.Parse(fixtureWorkItems[i]
+                    .Test.Properties.Get(PropertyNames.Order).ToString());
+
+                Assert.IsTrue(previousTestOrder < currentTestOrder);
+            }
+        }
+
+        private static readonly object[] Cases =
+        {
```

Comment:
I noticed while working on this that the test suite gets ordered alphabetically by default, and was so focused on making sure that I didn't add them alphabetically (so I could see the order change when it was converted into the WorkItems), that I didn't even notice I had added them in the "Order" order 😄  Oops!